cli/internal/graph/graph.go

```diff
@@ -76,12 +78,19 @@ func (g *CompleteGraph) GetPackageTaskVisitor(
 			return fmt.Errorf("Could not find definition for task")
 		}
 
+		// Task env mode is only independent when global env mode is `infer`.
+		taskEnvMode := globalEnvMode
+		if taskEnvMode == util.Infer && taskDefinition.PassthroughEnv != nil {
+			taskEnvMode = util.Strict
+		}
```

Comment:
Hmm, this seems like new logic? In the docs, we wrote:

> If strict mode is specified or inferred, all tasks are run in strict mode, regardless of their configuration.
[source](https://turbo.build/repo/docs/reference/command-line-reference#--experimental-env-mode)

Is this not true or intentional? (I thought I got this bit from either the code or a comment in Slack (I didn't come up with it))

Reply:
It's not new logic, it starts by inheriting `globalEnvMode`'s value which will be `strict` in the scenario described in the docs. The only thing it does is collapse the `infer` state for the task when it has not already been locked.

Reply:
I've added a new commit that collapses this further so that the output for a task is always `loose` or `strict`.